internal/data/common.go

```diff
@@ -108,118 +110,70 @@ func (a AttributeValue) BoolVal() bool {
 	return a.orig.BoolValue
 }
 
-func (a AttributeValue) SetString(v string) {
+func (a AttributeValue) SetStringVal(v string) {
+	a.copyValues(emptyAttributeKeyValue)
 	a.orig.Type = otlpcommon.AttributeKeyValue_STRING
 	a.orig.StringValue = v
```

Comment:
Alternatively it is possible to do:
```go
*a.orig = otlpcommon.AttributeKeyValue{Type:..., StringValue...}
```
This seems shorter and no need for copyValue call or for emptyAttributeKeyValue var.

Reply:
Strange. Let's keep your code, it is fine. This may be a frequent operation so let's not loose performance.